GLTFSDK/Source/SchemaValidation.cpp

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <GLTFSDK/SchemaValidation.h>
+#include <GLTFSDK/Exceptions.h>
+
+#include <sstream>
+#include <unordered_map>
+
+using namespace Microsoft::glTF;
+
+namespace
+{
+    class RemoteSchemaDocumentProvider : public rapidjson::IRemoteSchemaDocumentProvider
+    {
+    public:
+        RemoteSchemaDocumentProvider(std::unique_ptr<const ISchemaLocator> schemaLocator) : schemaLocator(std::move(schemaLocator))
+        {
+            assert(this->schemaLocator);
+        }
+
+        const rapidjson::SchemaDocument* GetRemoteDocument(const std::string& uri)
+        {
+            auto itDoc = schemaDocuments.find(uri);
+
+            if (itDoc != schemaDocuments.end())
+            {
+                return &(itDoc->second);
+            }
+
+            rapidjson::Document document;
+
+            if (document.Parse(schemaLocator->GetSchemaContent(uri)).HasParseError())
+            {
+                std::stringstream ss;
```

Comment:
Is this really better than:

`throw GLTFException("Schema document (" + uri + ") is not valid JSON");`

Reply:
I've replaced the use of stringstream with string::operator+ as this isn't performance critical code